### ISSUE TITLE
Should we really insist on a Version or Package-version header?

### ISSUE THREAD

**Issue by tarsius** [purcell/package-lint#93]
```
2 issues found:

1:1: warning: "Version:" or "Package-Version:" header is missing. MELPA will handle this, but other archives will not.
1:1: error: package.el cannot parse this buffer: Package lacks a "Version" or "Package-Version" header
```

I for one am not going to add it to my own packages. When they exist, these headers are usually outdated. `v1.0` contains `Version: 1.0`, but so do the 42 commits that follow until `v2.0` is released and the line is updated to `Version: 2.0`. No information is better than incorrect information. Of course you could update to `v1.0-git` in the first commit after the release. But wouldn't `package` choke on that too? And I would usually forget to do it anyway.

**Comment by purcell** [comment]
Yeah, this is true. I occasionally just add `;; Package-Version: 0` to my packages to assuage this complaint. The thing is, for single-file packages the lack of a version renders them incapable of installation using `package-install-file`, and (presumably) invalid for GNU ELPA, so that _is_ a concrete packaging issue, no?

**Comment by tarsius** [comment]
> The thing is, for single-file packages the lack of a version renders them incapable of installation using package-install-file, and (presumably) invalid for GNU ELPA, so that is a concrete packaging issue, no?

The former probably is, but the latter not - unless one actually is about to add a package to GNU Elpa of course ;-)

> I occasionally just add ;; Package-Version: 0 to my packages to assuage this complaint.

Melpa-Stable completely ignores that value right? In that case I could maybe be persuaded to add that to my headers.

**Comment by purcell** [comment]
> Melpa-Stable completely ignores that value right? In that case I could maybe be persuaded to add that to my headers.

Yes, both MELPA builds ignore all those version headers.

**Comment by Fanael** [comment]
> Of course you could update to `v1.0-git` in the first commit after the release. But wouldn't `package` choke on that too?

No, `;; Version: 1.0-git` is perfectly valid.

> The thing is, for single-file packages the lack of a version renders them incapable of installation using `package-install-file`

Yes, and this is why it's an error.

**Comment by tarsius** [comment]
> Yes, and this is why it's an error.

Okay then, to answer my own question: yes we should.
